bebraven/platform
# Add module state implementation, change proxy to mock

### Description

Part 2 of many towards removing LRS dependency. Based on #485.

Removes all mention of "LRS proxy", replacing it with an "LRS mock". Adds support for the state endpoint, alongside existing statement/interaction support. 

Note: This breaks projects until #482 is completed and merged. 

Task: https://app.asana.com/0/1174274412967132/1197477442813577/f

Testing:

1. Open a (working) Rise360 module assignment on Canvas.
2. Click through it, watch the states/statements get saved
3. Refresh the page, see that the state gets loaded and it picks up where you left off.



### Motivation and Context




### Screenshots (if appropriate):

<img width="1440" alt="Screen Shot 2020-12-02 at 1 11 56 PM" src="https://user-images.githubusercontent.com/1382374/100923170-69d85c00-34a4-11eb-8a57-184cde04c974.png">
<img width="1306" alt="Screen Shot 2020-12-02 at 12 41 09 PM" src="https://user-images.githubusercontent.com/1382374/100923171-69d85c00-34a4-11eb-84f0-324b02dfb31a.png">
<img width="1440" alt="Screen Shot 2020-12-02 at 4 45 07 PM" src="https://user-images.githubusercontent.com/1382374/100940614-ddd32e00-34bd-11eb-8164-387feb91e1c7.png">


### Types of changes




### Checklist:
